src/utils/Utils.ts

```diff
@@ -0,0 +1,6 @@
+export function truncate(str: string, n: number) {
```

Comment:
Please rename this function to something like `shortenTextWithEllipsis` or something. It is not doing truncation, so it's confusing to call it `truncate`.